src/CMakeLists.txt

```diff
@@ -0,0 +1,94 @@
+# Copyright (C) 2017-2020  The SymbiFlow Authors.
+#
+# Use of this source code is governed by a ISC-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/ISC
+#
+# SPDX-License-Identifier: ISC
+
+# minimum required CMAKE version
+CMAKE_MINIMUM_REQUIRED(VERSION 3.7 FATAL_ERROR)
+
+project(parse_fasm)
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../third_party/antlr4/runtime/Cpp/cmake)
+
+set(CMAKE_CXX_STANDARD 11)
+
+# add antrl4cpp artifacts to project environment
+include_directories(${ANTLR4_INCLUDE_DIRS})
+
+# set variable pointing to the antlr tool that supports C++
+# this is not required if the jar file can be found under PATH environment
+set(ANTLR_EXECUTABLE ${CMAKE_CURRENT_SOURCE_DIR}/../third_party/antlr4/antlr-4.9-complete.jar)
+
+# add macros to generate ANTLR Cpp code from grammar
+find_package(ANTLR REQUIRED)
+
+find_library(ANTLR4_RUNTIME NAMES antlr4-runtime REQUIRED)
+
+# Unit testing library
+add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../third_party/googletest EXCLUDE_FROM_ALL googletest)
+
+# Lexer and parser targets
+antlr_target(FasmLexer antlr/FasmLexer.g4 LEXER
+  OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  COMPILE_FLAGS -Xexact-output-dir
+  )
+antlr_target(FasmParser antlr/FasmParser.g4 PARSER VISITOR
+  OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  DEPENDS_ANTLR FasmLexer
+  COMPILE_FLAGS -Xexact-output-dir -lib ${ANTLR_FasmLexer_OUTPUT_DIR}
+  )
+
+# Explanation:
+# line 1: Skip lines starting in #define
+#      2: Extract TAGS(...) from dependencies
+#      3: Convert from TAGS('c', long_name) -> long_name = b'c, write to file
+add_custom_target(tags.py ALL
```

Comment:
Should this be a simple Python script rather than a bunch of greps and a sed?

Reply:
I don't think so. The complexity is in the regular expressions, which is unavoidable.